src/validation.cpp

```diff
@@ -4903,8 +4912,86 @@ bool CChainState::RollforwardBlock(const CBlockIndex* pindex, CCoinsViewCache& i
             pindex->GetBlockHash().ToString(), state.ToString());
     }
 
-    for (const CTransactionRef& tx : block.vtx) {
+    std::vector<std::pair<CAddressIndexKey, CAmount> > addressIndex;
+    std::vector<std::pair<CAddressUnspentKey, CAddressUnspentValue> > addressUnspentIndex;
+    std::vector<std::pair<CSpentIndexKey, CSpentIndexValue> > spentIndex;
+
+    for (unsigned int i = 0; i < block.vtx.size(); i++) {
+        const CTransactionRef &tx = block.vtx[i];
+        const uint256 txhash = tx->GetHash();
+
         if (!tx->IsCoinBase()) {
+            // Update indexes
+            if (fAddressIndex || fSpentIndex) {
+                for (size_t j = 0; j < tx->vin.size(); j++) {
+                    const CTxIn input = tx->vin[j];
+                    const Coin& coin = inputs.AccessCoin(tx->vin[j].prevout);
+                    const CTxOut &prevout = coin.out;
+                    uint160 hashBytes;
+                    int addressType;
+
+                    if (prevout.scriptPubKey.IsPayToScriptHash()) {
+                        hashBytes = uint160(std::vector<unsigned char>(prevout.scriptPubKey.begin()+2, prevout.scriptPubKey.begin()+22));
+                        addressType = 2;
```

Comment:
magic numbers?

Reply:
any options to reduce magic numbers in this PR?

Reply:
That's even less relevant to this PR than 76bc3830d562da35daddc5eb743e9703d0705da7 imo :D  Let's keep it `fix` focused.